federation/pkg/dnsprovider/providers/local/coredns/coredns.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package coredns is the implementation of pkg/dnsprovider interface for CoreDNS
+package coredns
+
+import (
+	"fmt"
+	etcdc "github.com/coreos/etcd/client"
+	"github.com/golang/glog"
+	"gopkg.in/gcfg.v1"
+	"io"
+	"k8s.io/kubernetes/federation/pkg/dnsprovider"
+	"strconv"
+	"strings"
+)
+
+// "local-coredns" should be used to use this DNS provider
+const (
+	ProviderName = "local-coredns"
```

Comment:
nit: This does not necessarily need to be local to the cluster, right?  As long as this provider can connect to the etcd endpoints, it should work?  Perhaps rename to just 'coredns'?